### DISCUSSION TITLE
Unable to create temporary file when uploading a file in a POST route

### DISCUSSION THREAD

**Discussion by koe01sense** [laravel/framework#53198] [Q&A]
How do I fix a warning wherein when I tried to upload a file, PHP or Laravel says `Warning: File upload error - unable to create a temporary file in Unknown on line 0`?

This is the configuration currently set on my php.ini
```
; Directory where the temporary files should be placed.
; Defaults to the system default (see sys_get_temp_dir)
;sys_temp_dir = "/tmp"
```
running the command above returns my temp directory
```
$ php artisan tinker
Psy Shell v0.12.4 (PHP 8.3.12 — cli) by Justin Hileman
> sys_get_temp_dir();
= "C:\Users\FULLSC~1\AppData\Local\Temp"
```

and creating tmpfile should return a boolean flag if the temp directory is not writeable but the result is opposite.
```
> $tempfile = tmpfile();
= stream resource #803
```

Here's the result of hitting the endpoint through postman.
![image](https://github.com/user-attachments/assets/efb338c7-dad0-48b6-acfd-267a3faddc63)

As for the controller method code, there's no particular usage yet for the uploaded file due to this warning
```
    public function uploadLogo(Workspace $workspace, Request $request) {
        // $request->validate([
        //     'logo' => ['file']
        // ]);

        $file = $request->file('logo');

        dd($file);

        return new WorkspaceResource($workspace);
    }
```

**Comment by ejsinfuego** [comment]
is your file upload enabled and also the file limit is enough in your `php.ini` config?

**Reply by koe01sense** [reply]
yes and yes. The file I uploaded is a small file size image around 1kb.

**Reply by ejsinfuego** [reply]
![image](https://github.com/user-attachments/assets/285c7dd8-b628-42e2-a675-0acaef9d26f0)

what's your current directory on that part of `php.ini` ? im thinking, there are two possible error. incorrect directory or authorization (write access)

**Reply by koe01sense** [reply]
I am using Laravel Herd and tried updating the said config using `C:\Temp` directory and now it works as expected. I'm confused with how it is unable to create a temporary file when the default temp directory `sys_get_temp_dir()`  for `upload_tmp_dir` config is a writeable directory.

**Reply by ejsinfuego** [reply]
I think it does, it's just that, you have incorrect directory.
